scipy/optimize/_differentialevolution.py

```diff
@@ -206,6 +207,17 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
 
         .. versionadded:: 1.9.0
 
+    vectorized : bool, optional
+        If ``vectorized is True``, `func` is sent an array of shape
+        ``(M, len(x))`` and is expected to return an array of shape ``(M,)``.
```

Comment:
`M` is specified in the `init` section as being the length of the population

Reply:
That's not what I'm picking at, though. I'm saying that `x` is defined as the first argument of `func`, and you also write that `func` is going to be sent an array of shape`(M, len(x))`. This would indicate that `x.shape == (M, len(x))`, which is only possible if `len(x)==M`. That's not what is meant (and I imagine most would understand what is intended), but interpreted literally, isn't that what it says?

Reply:
I've tried to do this in the definition for `func`.
N = number of parameters
C = number of constraints
M = number of solution vectors

Thus `func(x, *args)` with `x.shape == (N, M)` should result in an array of length `(M,)`.

Reply:
I've changed the nomenclature to be consistent with NonlinearConstraint
N = number of parameters
M = number of constraints
S = number of solution vectors